src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs

```diff
@@ -493,33 +501,6 @@ private static void CopySelectedItem(PackageSourceContextInfo selectedPackageSou
             Clipboard.SetText(selectedPackageSource.Source);
         }
 
-        private void PackageSourcesListBox_MouseUp(object sender, MouseEventArgs e)
```

Comment:
This change causes a minor change in behaviour. This custom mouse handler only toggled the checkbox when the checkbox is clicked.

WinForms' CheckedListBox behavior is:
* Clicking an item that is not selected will select it, but will not toggle the checked state
* Clicking an item that is already selected will select it, even if the click was on the text, not the checkbox
* Clicking any whitespace underneath the last item in the list will toggle the currently selected item's checkbox, as if the selected item itself was clicked

I don't have an opinion about the first two points, but I don't like the last one, clicking whitespace causing the checkbox to change. However, I've been unable to find how to change this behaviour and I don't feel it's important enough to spend more time investigating.

Reply:
Talking to someone from the WinForms team, the "clicking whitespace toggles the selected item's checkbox" comes from the win32 control behaviour, meaning it will be quite difficult indeed to change.